src/db/write.ts

```diff
@@ -296,20 +295,24 @@ export class Write extends Read {
     for (const [name, index] of this.indexes) {
```

Comment:
I think this is missing handling for indexes that are in basisIndexes but missing from this.indexes.